juju_verify/verifiers/ceph.py

```diff
@@ -109,60 +137,83 @@ def __hash__(self) -> int:
         """Return hash representation of AZ objects."""
         return hash(self.__str__())
 
-    def find_parent(self, node_id: int) -> Optional[NodeInfo]:
-        """Find the node that has the id in the children list."""
+    def get_node(self, name: str) -> NodeInfo:
+        """Get node by name."""
+        if name not in self._nodes_name_map.keys():
+            raise KeyError(f"Node {name} was not found.")
+
+        try:
+            node = self._nodes[self._nodes_name_map[name]]
+            assert node.name == name  # check that variable _nodes was not changed
+        except (IndexError, AssertionError) as error:
+            raise ValueError("Private value `_nodes` was changed.") from error
+
+        return node
+
+    def find_ancestor(
+        self, host_node: NodeInfo, required_type: str
+    ) -> Optional[NodeInfo]:
+        """Find ancestor, by using a replication rule."""
         for node in self._nodes:
-            if node.children and node_id in node.children:
+            if node.children and host_node.id in node.children:
+                if node.type != required_type:
+                    # NOTE (rgildein): it is necessary to go higher on the leaves of
+                    # the tree up to the desired leaf
```

Comment:
Please clarify this comment (eg. by giving an example). What are the possible values of `node.type` and `required_type`?